### PR TITLE
Split Button: Separate disabled capability for menu from primary action

### PR DESCRIPTION
**Pull request by montselozanod**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Right now a split button is disabled if the primary action of the split button is disabled. We want to separate this capability to enable scenarios where even though the primary action is disabled, we can still access the other menu options.

separate

### CODE REVIEW

**Comment by micahgodbolt** [comment]
Hey @montselozanod, this looks great.  My only concern is that we're technically breaking anyone who is expecting "disabled" to work on the entire button. @dzearing do you think it's worth leaving disabled as is and creating 2 separate xxxdisabled props? Or just communicate the change properly, as the fix is obviously quite easy.

**Comment by dzearing** [comment]
Yeah this feels a little strange to me. I'm not completely sure this would have been expected behavior and definitely would be a change. What exactly is the real-world use case? Are there alternative solutions to the use case?

**Comment by christiango** [comment]
I think I agree with @micahgodbolt , that disabled should disable both the menu launcher and the primary action for the sake of consistency with other components. We should definitely support disabling the split button primary action without disabling the menu, that is an expected scenario.

**Comment by betrue-final-final** [comment]
If we do this, and the menu isn't disabled, but the primary is, or vice versa, there needs to be clear and independent screen reader information for both functions. That may get a bit chatty. Why would this happen?

**Comment by dzearing** [comment]
I would be ok with just having a `primaryDisabled` flag which is used for this requirement.

**Comment by jspurlin** [comment]
The only aspect that is still unclear to me is how this control will be marked up when in this state... Maybe it essentially turns into a menuButton where enter, space, or alt+downArrow all expand the menu. Technically, the splitButton control is not disabled since it is still operable. The control should be disabled only when the whole control is not operable

File: packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx [hunk 1]
```diff
@@ -87,22 +88,24 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
          } = this.props;
 
     let { menuProps } = this.state;
+    // Button is disabled if the whole button (in case of splitbutton is disabled) or if the primary action is disabled
+    let isPrimaryButtonDisabled = (disabled || primaryDisabled);
```
Comment (christiango) [inline packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: nit: shouldn't need parens here.

**Pull request merged**